joomla/joomla-cms
# Cache Plugin improvements

### Description

After adding some options added for URI exclusions by @OctavianC in the cache plugin (https://github.com/joomla/joomla-cms/pull/7767), this PR adds some other improvements:
- Checks SEF and non SEF URI
- Since it also checks the non sef URI now you can exclude all pages of a
  component (ex: com_users)
- No need to backslash the slashes
- Do not cache or load from cache if site is offline
- Some other minor code improvements
- Show debugger even in cached pages
### How to test

With a fresh Joomla install with sample items, turn on cache plugin and exclude some SEF or Non-SEF page URI in "Exclude Pages" field, for instance:
- com_contact
- option=com_content
- /login
- /article-categories-view
1. Check if pages is being cached or not.
2. Put the site offline and see that it does not generates or load from cache.
### B/C

Should be ok.
### Notes

Since i'm not expert in Joomla code, if there are betters ways to do this, improvements are welcome.
